connectors/src/connectors/slack/bot.ts

```diff
@@ -593,34 +593,62 @@ async function answerMessage(
   if (buildContentFragmentRes.isErr()) {
     return buildSlackMessageError(buildContentFragmentRes);
   }
+
   let conversation: ConversationType | undefined = undefined;
   let userMessage: UserMessageType | undefined = undefined;
+
   if (lastSlackChatBotMessage?.conversationId) {
-    if (buildContentFragmentRes.value) {
-      const contentFragmentRes = await dustAPI.postContentFragment({
+    const ConversationNotFoundError = class extends Error {
```

Comment:
No classes/types within functions body.

Also why a custom error classes here?

Reply:
Also you can do something simpler for customer error classes.

```
const ConversationNotFoundError = class extends Error {
    readonly name = "ConversationNotFoundError";

    constructor(message: string) {
        super(message);
      }
}
```